File: packages/python-sdk/examples/oauth_bearer_payment_request.py
from payxcommerce import BearerTokenAuth, Client, ClientCredentials, Config
from payxcommerce.webhooks import event_types

credentials = ClientCredentials("YOUR_DEVELOPER_APP_CLIENT_ID", "YOUR_DEVELOPER_APP_CLIENT_SECRET")
access_token = credentials.access_token("payment_requests.write transactions.read balances.read refunds.write")
client = Client(Config(auth=BearerTokenAuth(access_token)))

response = client.payment_requests().create({
    "amount": 125.50,
    "currency": "USD",
    "purpose": "SDK bearer example order",
    "customer": {"name": "Jane Customer", "email": "customer@example.com", "country": "United States"},
    "webhook_url": "https://example.com/payxcommerce/webhook/order-1001",
    "ipn_events": event_types.default_subscriptions(),
    "metadata": {"source": "python-sdk-bearer-example"},
    "is_test": True,
})

print(response.get("checkout_url"))

File: packages/node-sdk/examples/oauth-bearer-payment-request.js
'use strict';

const { BearerTokenAuth, Client, ClientCredentials, Config, eventTypes } = require('../src');

(async () => {
  const credentials = new ClientCredentials('YOUR_DEVELOPER_APP_CLIENT_ID', 'YOUR_DEVELOPER_APP_CLIENT_SECRET');
  const accessToken = await credentials.accessToken('payment_requests.write transactions.read balances.read refunds.write');
  const client = new Client(new Config({ auth: new BearerTokenAuth(accessToken) }));
  const response = await client.paymentRequests().create({
    amount: 125.50,
    currency: 'USD',
    purpose: 'SDK bearer example order',
    customer: { name: 'Jane Customer', email: 'customer@example.com', country: 'United States' },
    webhook_url: 'https://example.com/payxcommerce/webhook/order-1001',
    ipn_events: eventTypes.defaultSubscriptions(),
    metadata: { source: 'node-sdk-bearer-example' },
    is_test: true
  });
  console.log(response.checkout_url);
})();
